URLConnection fails with JavaStaticSdk 11-ea+0

Follow up of https://github.com/gluonhq/client-maven-plugin/issues/216

After switching to JavaStatic 11-ea+0, (Http)URLConnection fails on Android and iOS.
```
[SUB] D/GraalCompiled(11552): Caused by: java.lang.UnsatisfiedLinkError: sun.security.ec.ECKeyPairGenerator.isCurveSupported([B)Z [symbol: Java_sun_security_ec_ECKeyPairGenerator_isCurveSupported or Java_sun_security_ec_ECKeyPairGenerator_isCurveSupported___3B]
[SUB] D/GraalCompiled(11552):   at com.oracle.svm.jni.access.JNINativeLinkage.getOrFindEntryPoint(JNINativeLinkage.java:153)
[SUB] D/GraalCompiled(11552):   at com.oracle.svm.jni.JNIGeneratedMethodSupport.nativeCallAddress(JNIGeneratedMethodSupport.java:57)
[SUB] D/GraalCompiled(11552):   at sun.security.ec.ECKeyPairGenerator.isCurveSupported(ECKeyPairGenerator.java)
```

The library `~/.gluon/substrate/javaStaticSdk/11-ea+0/ios-arm64/labs-staticjdk/lib/static/libsunec.a` contains `_JNI_OnLoad_sunec` and `_Java_sun_security_ec_ECKeyPairGenerator_isCurveSupported` symbols, however, these are not included with the current link command.

As a quick fix on iOS, moving `-lsunec` to `-Wl,-force_load,~/.gluon/substrate/javaStaticSdk/11-ea+0/ios-arm64/labs-staticjdk/lib/static/libsunec.a` solves the issue, as force_load includes all the symbols of the library.

--- a/src/main/resources/native/android/c/launcher.c
+++ b/src/main/resources/native/android/c/launcher.c
@@ -46,6 +46,7 @@ char *appDataDir;
 char *timeZone;
 
 int start_logger(const char *app_name);
+int JNI_OnLoad_sunec(JavaVM *vm, void *reserved);
 
 const char *origargs[] = {
     "myapp",
@@ -155,6 +156,9 @@ JNIEXPORT void JNICALL Java_com_gluonhq_helloandroid_MainActivity_startGraalApp(
     (*run_main)(argsize, graalArgs);
 
     LOGE(stderr, "called JavaMainWrapper_run\n");
+
+    // Invoke sunec
+    JNI_OnLoad_sunec(NULL, NULL);
 }
 
 // == expose window functionality to JavaFX native code == //

--- a/src/main/resources/native/android/c/launcher.c
+++ b/src/main/resources/native/android/c/launcher.c
@@ -46,6 +46,8 @@ char *appDataDir;
 char *timeZone;
 
 int start_logger(const char *app_name);
+
+// TODO: remove once https://github.com/oracle/graal/issues/2713 is fixed
 int JNI_OnLoad_sunec(JavaVM *vm, void *reserved);
 
 const char *origargs[] = {

--- a/src/main/resources/native/ios/AppDelegate.m
+++ b/src/main/resources/native/ios/AppDelegate.m
@@ -53,6 +53,7 @@ static __inline__ void gvmlog(NSString* format, ...)
 @end
 
 int startGVM(const char* userHome, const char* userTimeZone);
+int JNI_OnLoad_sunec(JavaVM *vm, void *reserved);
 
 extern int *run_main(int argc, const char* argv[]);
 
@@ -92,6 +93,9 @@ int main(int argc, char * argv[]) {
     NSString *tzName = [@"-Duser.timezone=" stringByAppendingString: [timeZone name]];
     const char *userTimeZone = [tzName UTF8String];
     startGVM(userHome, userTimeZone);
+
+    // Invoke sunec
+    JNI_OnLoad_sunec(nil, nil);
 }
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {

--- a/src/main/resources/native/ios/AppDelegate.m
+++ b/src/main/resources/native/ios/AppDelegate.m
@@ -53,6 +53,8 @@ static __inline__ void gvmlog(NSString* format, ...)
 @end
 
 int startGVM(const char* userHome, const char* userTimeZone);
+
+// TODO: remove once https://github.com/oracle/graal/issues/2713 is fixed
 int JNI_OnLoad_sunec(JavaVM *vm, void *reserved);
 
 extern int *run_main(int argc, const char* argv[]);

